docs/admin/kubeadm.md

```diff
@@ -460,6 +520,91 @@ started.
+The kubeadm discovery system has several options, each with security tradeoffs.
+The right method for your environment depends on how you provision nodes and the
+security expectations you have about your network and node lifecycles.
+
+### Token-based discovery with CA pinning
+_This is the default mode in Kubernetes 1.8._ In this mode, kubeadm downloads 
+the cluster configuration (including root CA) and validates it using the token
+as well as validating that the root CA public key matches the provided hash and
+that the API server certificate is valid under the root CA.
+
+**Example `kubeadm join` command:**
+
+ - `kubeadm join --discovery-token abcdef.1234567890abcdef --discovery-token-ca-cert-hash sha256:1234..cdef 1.2.3.4:6443`
+
+**Advantages:**
+
+ - Allows bootstrapping nodes to securely discover a root of trust for the
+   master even if other worker nodes or the network are compromised.
+
+ - Convenient to execute manually since all of the information required fits
+   into a single `kubeadm join` command that is easy to copy and paste.
+
+**Disadvantages:**
+
+ - The CA hash is not normally known until the master has been provisioned,
+   which can make it more difficult to build automated provisioning tools that
+   use kubeadm.
+
+### Token-based discovery without CA pinning
+_This was the default in Kubernetes 1.7 and earlier_, but comes with some
+important caveats. This mode relies only on the symmetric token to sign
+(HMAC-SHA256) the discovery information that establishes the root of trust for
+the master. It's still possible in Kubernetes 1.8 and above using the
+`--discovery-token-unsafe-skip-ca-verification` but you should consider using
+one of the other modes if possible.
+
+**Example `kubeadm join` command:**
+
+ - `kubeadm join --discovery-token abcdef.1234567890abcdef --discovery-token-unsafe-skip-ca-verification 1.2.3.4:6443`
+
+**Advantages:**
+
+ - Still protects against many network-level attacks.
+
+ - The token can be generated ahead of time and shared with the master and
+   worker nodes, which can then bootstrap in parallel without coordination. This
+  allows it to be used in many provisioning scenarios.
+
+**Disadvantages:**
+
+ - If an attacker is able to steal a bootstrap token via some vulnerability,
+   they can use that token (along with network-level access) to impersonate the
+   master to other bootstrapping nodes. This may or may not be an appropriate
+   tradeoff in your environment. 
+
+### File or HTTPS-based discovery
+This provides an out-of-band way to establish a root of trust between the master
+and bootstrapping nodes.   Consider using this mode if you are building automated provisioning
+using kubeadm. 
+
+**Example `kubeadm join` commands:**
+
+ - `kubeadm join --discovery-file path/to/file.conf` (local file)
+
+ - `kubeadm join --discovery-file https://url/file.conf` (remote HTTPS URL)
+
+**Advantages:**
+ - Allows bootstrapping nodes to securely discover a root of trust for the
```

Comment:
nice! last nit: you'll want to put an extra space here (and in the next bullet point below), otherwise the formatting won't register

Reply:
Thanks. I had just tried to fix that but I missed this one.